### PR TITLE
FCM support, fixed HTTP crash, code style

### PR DESCRIPTION
**Pull request by dzolnai**
* Removed C2DM support, added Firebase Cloud Messaging support. Since it is an extension of GCM, we can still use the GCM type against the Tiqr server, and it will work as expected.
* Fixed an issue where the app would crash if it would contact a server over plain HTTP instead of HTTPS.
* Fixed code style issue from previous PR, removed support-v4 lib since it was unused and overruled by appcompat.

### CODE REVIEW

File: shared/src/main/java/org/tiqr/service/authentication/AuthenticationService.java [hunk 1]
```diff
@@ -233,17 +231,17 @@ protected AuthenticationError doInBackground(Void... voids) {
...
-
-                    String response = Utils.urlConnectionResponseAsString(httpsURLConnection);
-                    String versionHeader = httpsURLConnection.getHeaderField("X-TIQR-Protocol-Version");
+                    HttpURLConnection httpURLConnection = (HttpURLConnection)authenticationURL.openConnection();
+                    httpURLConnection.setRequestMethod("POST");
+                    httpURLConnection.setRequestProperty("ACCEPT", "application/json");
+                    httpURLConnection.setRequestProperty("X-TIQR-Protocol-Version", Constants.PROTOCOL_VERSION);
+                    httpURLConnection.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
+                    httpURLConnection.setRequestProperty("Content-Length", String.valueOf(postData.length));
+                    httpURLConnection.setDoOutput(true);
```
Comment (andrei-egeniq) [inline shared/src/main/java/org/tiqr/service/authentication/AuthenticationService.java]: setDoOutput caused issues with some special urls in BNR. I think the pipe char `|`

Comment (andrei-egeniq) [inline shared/src/main/java/org/tiqr/service/authentication/AuthenticationService.java]: But it might be an GET only issue

Comment (dzolnai) [inline shared/src/main/java/org/tiqr/service/authentication/AuthenticationService.java]: You are not supposed to write to a GET request, so I guess that was the problem :)

**Comment by andrei-egeniq** [comment]
Looks ok, no merge rights.

**Pull request merged**